app/code/Magento/Backend/view/adminhtml/templates/store/switcher.phtml

```diff
@@ -8,222 +8,132 @@
-                $block->getAuthorization()->isAllowed('Magento_Backend::store')): ?>
-                <li class="dropdown-toolbar">
-                    <a href="<?= /* @noEscape */ $block->getUrl('*/system_store');
-                    ?>"><?= $block->escapeHtml(__('Stores Configuration')) ?></a>
-                </li>
-            <?php endif; ?>
-        </ul>
+                <?php if ($block->getShowManageStoresLink() &&
+                    $block->getAuthorization()->isAllowed('Magento_Backend::store')): ?>
+                    <li class="dropdown-toolbar">
+                        <a href="<?= /* @noEscape */ $block->getUrl('*/system_store');
+                        ?>"><?= $block->escapeHtml(__('Stores Configuration')) ?></a>
+                    </li>
+                <?php endif; ?>
+            </ul>
+        </div>
+        <?= $block->getHintHtml() ?>
     </div>
-    <?= $block->getHintHtml() ?>
-</div>
-
-    <?php
-    $useConfirm = (int)$block->getUseConfirm();
-    $scriptString = <<<script
-require([
-    'jquery',
-    'Magento_Ui/js/modal/confirm'
-], function(jQuery, confirm){
-
-    (function($) {
-        var storesList = $('[data-role=stores-list]');
-        storesList.on('click', '[data-value]', function(event) {
-            var val = $(event.target).data('value');
-            var role = $(event.target).data('role');
-            var switcher = $('[data-role='+role+']');
-
-            event.preventDefault();
-
-            if (!switcher.val() || val != switcher.val()) {
-                switcher.val(val).trigger('change'); // Set the value & trigger event
-            }
-        });
-    })(jQuery);
-
-    var scopeSwitcherHandler;
 
-    function switchScope(obj) {
-        var switcher = jQuery(obj);
-        var scopeId = switcher.val();
-        var scopeParams = '';
-        if (scopeId) {
-            scopeParams = switcher.data('param') + '/' + scopeId + '/';
-        }
-
-        if (obj.switchParams) {
-            scopeParams += obj.switchParams;
-        }
-
-        if ((typeof scopeSwitcherHandler) != 'undefined') {
-            var switcherParams = {
-                scopeId: scopeId,
-                scopeParams: scopeParams,
-                useConfirm: {$useConfirm}
-            };
-            scopeSwitcherHandler(switcherParams);
-        } else {
-script;
-    if ($block->getUseConfirm()) {
-                $scriptString .= '
-            confirm({
-                content:  "' . $block->escapeJs(__(
-                    'Please confirm scope switching. All data that hasn\'t been saved will be lost.'
-                )) . '",
-                actions: {
-                    confirm: function() {
-                        reload();
-                    },
-                    cancel: function() {
-                        obj.value = \'' . $block->escapeJs($block->getStoreId()) . '\';
-                    }
+    <script type="text/x-magento-init">
+        {
+            ".store-switcher": {
```

Comment:
Looks like it should be one global service,
 ```suggestion
            "*": {
```

Reply:
Corrected that too